qutebrowser/completion/completer.py

```diff
@@ -204,25 +204,18 @@ def _get_new_completion(self, parts, cursor_part):
             return sortfilter.CompletionFilterModel(source=model, parent=self)
         # delegate completion to command
         try:
-            completions = cmdutils.cmd_dict[parts[0]].completion
+            cmd = cmdutils.cmd_dict[parts[0]]
         except KeyError:
             # entering an unknown command
             return None
-        if completions is None:
-            # command without any available completions
-            return None
-        dbg_completions = [c.name for c in completions]
         try:
             idx = cursor_part - 1
-            completion = completions[idx]
+            completion = cmd.get_pos_arg_info(idx).completion
         except IndexError:
-            # More arguments than completions
-            log.completion.debug("completions: {}".format(
-                ', '.join(dbg_completions)))
+            # user provided more positional arguments than the command takes
+            return None
+        if completion is None:
             return None
-        dbg_completions[idx] = '*' + dbg_completions[idx] + '*'
-        log.completion.debug("completions: {}".format(
-            ', '.join(dbg_completions)))
```

Comment:
I'm fine with leaving them out and adding new ones when we feel we actually need them.